app/controllers/posts_controller.rb

```diff
@@ -1,16 +1,17 @@
 class PostsController < ApplicationController
-  before_action :authenticate_user!
-
-  before_action :set_post, only: %i[ show edit update destroy ]
+  before_action :authenticate_user!, only: [:index, :show, :new, :create, :edit, :update, :destroy]
+  before_action :set_post, only: [:show, :edit, :update, :destroy]
+  before_action :check_post_owner, only: [:edit, :update, :destroy]
 
   # GET /posts or /posts.json
   def index
     @posts = Post.all
   end
 
   # GET /posts/1 or /posts/1.json
-  def show
+  def show 
     @post = Post.find(params[:id])
```

Comment:
Since your before action for set_post is used for show, this shouldn't be needed here as the action should already have access to the instance variable for @post.